src/In.ProjectEKA.HipService/Gateway/GatewayClient.cs

```diff
@@ -2,39 +2,94 @@ namespace In.ProjectEKA.HipService.Gateway
-            this.centralRegistryClient = centralRegistryClient;
-            this.configuration = gatewayConfiguration;
+            configuration = gatewayConfiguration;
+        }
+
+        public virtual async Task<Option<string>> Authenticate()
+        {
+            try
+            {
+                var json = JsonConvert.SerializeObject(new
+                {
+                    clientId = configuration.ClientId,
+                    clientSecret = configuration.ClientSecret
+                }, new JsonSerializerSettings
+                {
+                    NullValueHandling = NullValueHandling.Ignore,
+                    ContractResolver = new DefaultContractResolver
+                    {
+                        NamingStrategy = new CamelCaseNamingStrategy()
+                    }
+                });
+                var message = new HttpRequestMessage
+                {
+                    RequestUri = new Uri($"{configuration.Url}/v1/sessions"),
+                    Method = HttpMethod.Post,
+                    Content = new StringContent(json, Encoding.UTF8, MediaTypeNames.Application.Json)
+                };
+
+                var responseMessage = await httpClient.SendAsync(message).ConfigureAwait(false);
+                var response = await responseMessage.Content.ReadAsStringAsync();
+
+                if (!responseMessage.IsSuccessStatusCode)
+                {
+                    var error = await responseMessage.Content.ReadAsStringAsync();
+                    Log.Error(
+                        $"Failure in getting the token with status code {responseMessage.StatusCode} {error}");
+                    return Option.None<string>();
+                }
+
+                var definition = new {accessToken = "", tokenType = ""};
+                var result = JsonConvert.DeserializeAnonymousType(response, definition);
+                return Option.Some($"{result.tokenType} {result.accessToken}");
+            }
+            catch (Exception exception)
+            {
+                Log.Error(exception, exception.StackTrace);
+                return Option.None<string>();
+            }
         }
 
         public virtual async Task SendDataToGateway<T>(string urlPath, T response, string cmSuffix)
         {
-            await PostTo(configuration.Url + urlPath, response, cmSuffix)
-                .ConfigureAwait(false);
+            await PostTo(configuration.Url + urlPath, response, cmSuffix).ConfigureAwait(false);
         }
 
         private async Task PostTo<T>(string gatewayUrl, T representation, string cmSuffix)
         {
             try
             {
-                var token = await centralRegistryClient.Authenticate();
-                token.MatchSome(async accessToken => await httpClient
-                    .SendAsync(CreateHttpRequest(gatewayUrl, representation, accessToken, cmSuffix))
-                    .ConfigureAwait(false));
+                var token = await Authenticate();
```

Comment:
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5754537)